packages/mobile/src/exchange/WithdrawCeloScreen.tsx

```diff
@@ -51,6 +57,25 @@ function WithdrawCeloScreen({ navigation }: Props) {
     CURRENCY_ENUM.GOLD
   )
 
+  useEffect(() => {
+    getSendFee(
+      accountAddress,
+      CURRENCY_ENUM.GOLD,
+      {
+        recipientAddress: RANDOM_ADDRESS,
+        amount: goldBalanceNumber,
+        comment: '',
+      },
+      false
+    )
+      .then((estimate: BigNumber) => {
+        setFeeEstimate(divideByWei(estimate))
+      })
+      .catch((error) => {
+        Logger.warn('WithdrawCeloScreen', `Error found while estimating gas fee: ${error}`)
+      })
+  }, [])
+
```

Comment:
Async functions should be avoided in general in `useEffect` because they can trigger actions after the component has been unmounted.

`useAsync` from [react-async-hook](https://github.com/slorber/react-async-hook) solves this.

Now we have [CalculateFee](https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/fees/CalculateFee.tsx) which uses `useAsync` under the hood and mostly does what you want here.

I originally created it as a component because most components we had at the time, were class components.
But we could export a `useSendFee` hook [from it now](https://github.com/celo-org/celo-monorepo/blob/e67eba1cc9a54cdc6ed38425eea98a35e743aa6f/packages/mobile/src/fees/CalculateFee.tsx#L92-L111).

Added benefit is it provides loading/error states so we can display a spinner while it's calculating and it displays an error if it fails.

Reply:
Thanks for this! I had run into this many times and always solved it in this inelegant way, the `useAsync` hook looks great.